runbotics/runbotics-actions-windows/src/automation/office/excel.utils.ts

```diff
@@ -2,22 +2,33 @@ import { StartCellCoordinates } from "./excel.types";
 import ExcelErrorLogger from "./excelError.logger";
 
 export class ExcelUtils {
-    static switchWorksheet(getSession: () => any, setSession: (session: any) => void, setPrevWorksheet: (worksheet?: string) => void, worksheet?: string): void {
+    static switchWorksheet(getSession: () => any, setPrevWorksheet: (worksheet?: string) => void, worksheet?: string): void {
         const session = getSession();
         if (!session) return;
         if (!worksheet || worksheet === session.ActiveSheet.Name) return;
 
         setPrevWorksheet(session.ActiveSheet.Name);
         session.Worksheets(worksheet).Activate();
-        setSession(session);
     }
 
-    static switchPrevWorksheet(getSession: () => any, setSession: (session: any) => void, setPrevWorksheet: (worksheet: string | null) => void, prevWorksheet: string | null): void {
+    static switchPrevWorksheet(getSession: () => any, setPrevWorksheet: (worksheet: string | null) => void, prevWorksheet: string | null): void {
         if (!prevWorksheet) return;
         const session = getSession();
         session.Worksheets(prevWorksheet).Activate();
         setPrevWorksheet(null);
-        setSession(session);
+    }
+
+    static checkIfWorksheetExist(getSession: () => any, worksheet: string): boolean {
+        const session = getSession();
+        const worksheets = session.Worksheets;
+        const worksheetUpper = worksheet.toUpperCase();
+        for (let i = 1; i <= worksheets.Count; i++) {
+            const sheet = worksheets(i);
+            if (sheet.Name.toUpperCase() === worksheetUpper) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
just use `Array.prototype.some`

Reply:
it doesn't work in this case :(

Reply:
try
const worksheets = this.session().Worksheets;
  return Array
    .from(worksheets)
    .some(
      sheet => sheet.Name.toUpperCase() === worksheet.toUpperCase()
    );

Reply:
unfortunately, still not :(
![image](https://github.com/runbotics/runbotics/assets/63171496/405ea448-8eeb-41b6-b7e8-ed28707ed5d5)